graphql-server/src/index.ts

```diff
@@ -39,14 +54,202 @@ async function main() {
   app.use(express.static(path.resolve(__dirname, "../build")));
   app.use(bodyParser.urlencoded({ extended: true }));
   app.use(express.json({ limit: "50mb" })); // increase to support bulk searching
+  app.use(
+    cors({
+      origin: "https://localhost:3006",
+      credentials: true,
+    })
+  );
+
+  const keycloakIssuer = await Issuer.discover(
+    "https://smile-dev.mskcc.org:8443/realms/smile"
+  );
+
+  const keycloakClient = new keycloakIssuer.Client({
+    client_id: props.keycloak_client_id,
+    client_secret: props.keycloak_client_secret,
+    redirect_uris: ["https://localhost:4000/auth/callback"],
+    response_types: ["code"],
+  });
+
+  const memoryStore = new expressSession.MemoryStore();
+  app.use(
+    expressSession({
+      secret: props.express_session_secret,
+      resave: false,
+      saveUninitialized: true,
+      store: memoryStore,
+    })
+  );
+
+  app.use(passport.initialize());
+
+  // Enables persistent login sessions; equivalent to `app.use(passport.authenticate('session'))`
+  app.use(passport.session());
+
+  // These two functions are required for the Session strategy above to work
+  passport.serializeUser(function (user: any, done: any) {
+    done(null, user);
+  });
+  passport.deserializeUser(function (user: any, done: any) {
+    done(null, user);
+  });
+
+  let ssoSessionIdleTimeout = 0;
+  let loginTime = 0;
+  let id_token = "";
+
+  passport.use(
+    "oidc",
+    new Strategy(
+      { client: keycloakClient },
+      (tokenSet: any, userinfo: any, done: any) => {
+        ssoSessionIdleTimeout = tokenSet.refresh_expires_in * 1000; // convert to ms because it's in s by default
+        id_token = tokenSet.id_token;
+        return done(null, tokenSet.claims());
+      }
+    )
+  );
+
+  app.get("/login", (req, res, next) => {
```

Comment:
this stuff will all disappear when we follow up with graphql implementation, but in general we should avoid dumping alll this stuff into the "main" file (index.ts).  Right now it's just a mishmash of things.

Reply:
Agreed. A lot of this stuff will change with the GraphQL implementation, so I didn't want to prematurely refactor/organize this code into helper files.